### PR TITLE
Support timezones without daylight savings time

### PR DESCRIPTION
**Pull request by mderoy**

### CODE REVIEW

File: mcs/class/corlib/System/TimeZoneInfo.Unity.cs [hunk 1]
```diff
@@ -1,36 +1,13 @@
...
-// 
-// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
-// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
-// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
-// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
-// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
-// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//	Michael DeRoy
+//	Jonathan Chambers
```
Comment (joncham) [inline mcs/class/corlib/System/TimeZoneInfo.Unity.cs]: Might want emails

File: mcs/class/corlib/System/TimeZoneInfo.Unity.cs [hunk 2]
```diff
@@ -45,90 +22,100 @@ namespace System {
...
+			return rule;
+		}
+
 		static TimeZoneInfo CreateLocalUnity ()
 		{
 			Int64[] data;
-		    string[] names;
-		    if (!System.CurrentSystemTimeZone.GetTimeZoneData (1973, out data, out names))
+			string[] names;
+			int currentYear = DateTime.UtcNow.Year;
```
Comment (joncham) [inline mcs/class/corlib/System/TimeZoneInfo.Unity.cs]: No chance of infinite recursion here, right?

File: mcs/class/corlib/System/TimeZoneInfo.Unity.cs [hunk 3]
```diff
@@ -45,90 +22,100 @@ namespace System {
...
-					continue;
-				
-				DaylightTime dlt = new DaylightTime (new DateTime (data[(int)TimeZoneData.DaylightSavingStartIdx]),
-								     new DateTime (data[(int)TimeZoneData.DaylightSavingEndIdx]),
-								     new TimeSpan (data[(int)TimeZoneData.AdditionalDaylightOffsetIdx]));
-				
-				DateTime dltStartTime = new DateTime(1, 1, 1).Add(dlt.Start.TimeOfDay);
-				DateTime dltEndTime = new DateTime(1, 1, 1).Add(dlt.End.TimeOfDay);
+			TimeSpan utcOffsetTS = TimeSpan.FromTicks(data[(int)TimeZoneData.UtcOffsetIdx]);
+			char utcOffsetSign = (utcOffsetTS >= TimeSpan.Zero) ? '+' : '-';
```
Comment (joncham) [inline mcs/class/corlib/System/TimeZoneInfo.Unity.cs]: `var` please

Comment (mderoy) [reply mcs/class/corlib/System/TimeZoneInfo.Unity.cs]: ok ;(

File: mcs/class/corlib/System/TimeZoneInfo.Unity.cs (hunk 2 above)
Comment (mderoy) [reply mcs/class/corlib/System/TimeZoneInfo.Unity.cs]: No, looks like DateTime.UtcNow calls into an icall to get the current system time..converts that to ticks and passes it into the DateTime(<ticks>) constructor...should be no change of recursion, and I'm not seeing it myself.

**Pull request merged**